COMMIT-GUIDELINES.md

```diff
@@ -0,0 +1,161 @@
+# 5.0.0-alpha.3 (2024-12-17)
+
+### Bug Fixes
+
+- **grid.mixins:** declare $modules before use 1c7791f
+- **utils:** isNumber function validation inverted 37637b6
+- **utils:** update `retrieveEnumKeys` function 325d3af
+```
+
+#### Breaking changes
+
+A breaking change is a change that is significant enough to cause an incompatibility with the current state of the packages. This could be a removed component property, a set of styles that work differently, or a hole host of different changes.
+
+Breaking changes are denoted in the footer of the commit by starting it with `BREAKING CHANGE:` followed by a message about what the breaking change impacts.
+
+It is highly recommended to add a clear reason for the breaking change and what actions need to be taken when upgrading to the newer version. See the example under [Breaking change](#breaking-change) for an idea of how to communicate the change.
+
+It is important to note: Breaking changes can be added to any commit irregardless of the commit prefix, they are not just relegated to `fix` or `feat` commit prefixes as those aren't always relevant when introducing a breaking change.
+
+## Examples
+
+Good commit messages go along way in creating and maintaining a high quality project. They tell the story of how it grew and evolved overtime. They also help when investigating where things happened, should you ever have to go back in time. Commit messages are almost as important as high quality documentation and code comments.
+
+### Base structure
+
+The base structure of a commit message is:
+
+```
+prefix(optional-scope): subject
+
+body
+
+footer
+```
+
+### Commits with prefix and scope
+
+Adding a prefix and scope helps indicate the impact and theme of the commit and what it impacts.
+
+Commits can be simple:
+
+```
+chore(monorepo): add useWorkspaces to lerna.json
+```
+
+Commits can be more detailed:
+
+```
+fix(utils): isNumber function validation inverted
+
+This fix handles validating numbers are numbers and not numbers are `NaN`.
+
+- Add `undefined` and `null` checks to `isNumber` to guarantee they're
+  false.
+- Add unit tests to validate this case via the `date-validation-utils` function `isInvalidYear`.
+- Add unit tests to validate the `isNumber` util function.
+
+JIRA: DS-1234
+```
+
+Commits can have body text span multiple lines as paragraphs using new lines:
+
+```
+fix(utils): update retrieveEnumKeys function
+
+Updates to `isNumeric` invalided the logic this function was using,
+inverting it restores intended functionality.
+
+- Add unit tests to validate different types of enum cases as well as
+  `null` and `undefined`.
+```
+
+### Breaking change
+
+Adding `BREAKING CHANGE:` to the footer denotes the commit has a breaking change and that there should be a major version increment.
+
+```
+feat(ontario-card)!: added layout prop
+ - Updated unit tests
+ - Regenerated component definitions
+
+BREAKING CHANGE: `card-type` prop is now deprecated. Use `layout` prop instead.
+```
```

Comment:
Similar to my note in the bigger Breaking Change section above, is it worthwhile to put a note about the exclamation mark at the end of a commit prefix as a best practice?

e.g. It is also a best practice to include an exclamation mark after your commit prefix/scope. This is not used as an identifier by the changelog tooling (instead the BREAKING CHANGE footer identifier is leveraged), but it can be used as a further identifier to the reader that the change is of a breaking nature.